src/loaders/svg/tvgSvgLoader.cpp

```diff
@@ -2698,6 +2569,367 @@ static GradientFactoryMethod _findGradientFactory(const char* name)
+{
+    for (uint32_t i = 0; i < src.count; ++i) {
+        dst.push(src.data[i]);
+    }
+}
+
+
+static void _inheritGradient(SvgLoaderData* loader, SvgStyleGradient* to, SvgStyleGradient* from)
+{
+    if (!to || !from) return;
+
+    if (!((int)to->flags & (int)SvgGradientFlags::SpreadMethod) &&
+        ((int)from->flags & (int)SvgGradientFlags::SpreadMethod)) {
+        to->spread = from->spread;
+        to->flags = (SvgGradientFlags)((int)to->flags | (int)SvgGradientFlags::SpreadMethod);
+    }
+    bool gradUnitSet = (int)to->flags & (int)SvgGradientFlags::GradientUnits;
+    if (!((int)to->flags & (int)SvgGradientFlags::GradientUnits) &&
+        ((int)from->flags & (int)SvgGradientFlags::GradientUnits)) {
+        to->userSpace = from->userSpace;
+        to->flags = (SvgGradientFlags)((int)to->flags | (int)SvgGradientFlags::GradientUnits);
+    }
+
+    if (!to->transform && from->transform) {
+        to->transform = (Matrix*)malloc(sizeof(Matrix));
+        if (to->transform) memcpy(to->transform, from->transform, sizeof(Matrix));
+    }
+
+    if (to->type == SvgGradientType::Linear && from->type == SvgGradientType::Linear) {
+        for (unsigned int i = 0; i < sizeof(linear_tags) / sizeof(linear_tags[0]); i++) {
+            bool coordSet = (int)to->flags & (int)linear_tags[i].flag;
+            if (!((int)to->flags & (int)linear_tags[i].flag) && ((int)from->flags & (int)linear_tags[i].flag)) {
+                linear_tags[i].tagInheritHandler(to, from);
+            }
+
+            //GradUnits not set directly, coord set
+            if (!gradUnitSet && coordSet) {
+                linear_tags[i].tagRecalc(loader, to->linear, to->userSpace);
+            }
+            //GradUnits set, coord not set directly
+            if (to->userSpace == from->userSpace) continue;
+            if (gradUnitSet && !coordSet) {
+                linear_tags[i].tagInheritedRecalc(loader, to->linear, to->userSpace);
+            }
+        }
+    } else if (to->type == SvgGradientType::Radial && from->type == SvgGradientType::Radial) {
+        for (unsigned int i = 0; i < sizeof(radialTags) / sizeof(radialTags[0]); i++) {
+            bool coordSet = (int)to->flags & (int)radialTags[i].flag;
+            if (!((int)to->flags & (int)radialTags[i].flag) && ((int)from->flags & (int)radialTags[i].flag)) {
+                radialTags[i].tagInheritHandler(to, from);
+            }
+
+            //GradUnits not set directly, coord set
+            if (!gradUnitSet && coordSet) {
+                radialTags[i].tagRecalc(loader, to->radial, to->userSpace);
+            }
+            //GradUnits set, coord not set directly
+            if (to->userSpace == from->userSpace) continue;
+            if (gradUnitSet && !coordSet) {
+                radialTags[i].tagInheritedRecalc(loader, to->radial, to->userSpace);
+            }
+        }
+    }
+
+    if (to->stops.count == 0) _cloneGradStops(to->stops, from->stops);
```

Comment:
In the same context as now, can we add `stops` flag?
The reason for adding the flag is to check whether it is set to 0 or not set.
I think `to->stops.count == 0` is ambiguous.
Of course, it seem to be a no problem now. (Maybe?)